File: ruby/5k-extract-the-domain-name-from-an-url/domain_name.rb
# https://www.codewars.com/kata/514a024011ea4fb54200004b
# Extract the domain name from a string.

def domain_name(url)
  url.sub(/https?:\/\//, "")
     .sub(/www\./, "")
     .sub(/\..+/, "")
end

# ---------------------- #
# Tests
# ---------------------- #
def test(input, expected)
  puts "Testing with input #{input}"
  result = domain_name(input)

  if result == expected
    puts " => Test passed!\n\n"
  else
    puts " => Test failed. :("
    puts "    Got:      #{result.inspect}"
    puts "    Expected: #{expected.inspect}\n\n"
  end
end

test "http://google.com",   "google"
test "http://google.co.jp", "google"
test "www.xakep.ru",        "xakep"
test "https://youtube.com", "youtube"

File: js/5k-extract-domain-name-from-url/domainName.js
// https://www.codewars.com/kata/514a024011ea4fb54200004b/
// Extract the domain name from a URL.
//
function domainName(url){
    return url.replace(/https?:\/\//g, "")
              .replace("www.", "")
              .replace(/\..+/g, "")

    // Alternative solution
    // return url.replace(/(https?:\/\/)?(www\.)?/g, "").split(".")[0]
}

// ----------------------- //
// Tests
// ----------------------- //
function test(input, expected) {
    result = domainName(input)
    
    console.log("Testing with " + input + ".")
    if (result === expected) {
        console.log("Test passed!")
    } else {
        console.log("Test failed. :(")
        console.log(" => Got:      " + result)
        console.log("    Expected: " + expected)
    }
    console.log()
}

test("http://google.com", "google");
test("http://google.co.jp", "google");
test("www.xakep.ru", "xakep");
test("https://youtube.com", "youtube");
